dataset-spec/template.schema.json

```diff
@@ -142,88 +204,123 @@
+                                                "viridis_r",
+                                                "turbo",
+                                                "prism",
+                                                "Spectral"
                                             ]
-                                        },
-                                        "remarks": {
-                                            "type": "string"
-                                        }
+                                        ]
+                                    },
+                                    "color_value_min": {
+                                        "type": "number",
+                                        "description": "minimum value for colour scale",
+                                        "examples": [
+                                            -10500,
+                                            0,
+                                            1.735e-9,
+                                            4.5e6
+                                        ]
+                                    },
+                                    "color_value_max": {
+                                        "type": "number",
+                                        "description": "maximum value for colour scale",
+                                        "examples": [
+                                            -9500,
+                                            0,
+                                            2.978e-9,
+                                            8.5e6
+                                        ]
                                     }
                                 }
-                             },
-                             "color_bar_name": {
-                                 "type": "string",
-                                 "description": "name of a matplotlib colormap: https://matplotlib.org/stable/tutorials/colors/colormaps.html",
-                                 "examples": [
-                                     ["viridis_r", "turbo", "prism", "Spectral"]
-                                 ]
-                             },
-                             "color_value_min": {
-                                 "type": "number",
-                                 "description": "minimum value for colour scale",
-                                 "examples": [-10500, 0, 1.735e-9, 4.5e6]
-                             },
-                             "color_value_max": {
-                                 "type": "number",
-                                 "description": "maximum value for colour scale",
-                                 "examples": [-9500, 0, 2.978e-9, 8.5e6]
-                             },
-                             "metadata": {
-                                 "type": "object"
+                            }
+                        }
+                    }
+                },
+                "sources": {
```

Comment:
`sources` is not modelled correctly. We have two options for sources, either we use an xcube data store, then `store_id` is required _or_ we manually download data and open it locally, then `download_url` is required. Each option has its respective set of additional applicable properties.

Reply:
Now enforced with a `oneOf` operator in the schema.